File: wtt_web_client/src/models/Actions/Abstract/ADbAction.ts
import ADbObjectWithGuid from "src/models/Common/ADbObjectWithGuid";

export default interface ADbAction extends ADbObjectWithGuid
{
	type: string;
	name: string;
	description: string;
	next: string | null;
	continueExecutionInCaseOfCriticalError: boolean;
	columnId: number;
	inColumnId: number;
}

File: wtt_main_server/wtt_main_server_data_models/Database/Abstract/ADbAction.cs
using DbConcept;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace wtt_main_server_data.Database.Abstract;

public abstract class ADbAction : ObjectWithGuid
{
	public abstract string Type { get; } 

	public string Name { get; set; } = "Action";
	public string Description { get; set; } = "";
	public Guid? Next { get; set; }
	public bool ContinueExecutionInCaseOfCriticalError { get; set; } = false;

	public int ColumnId { get; set; } = 0;
	public int InColumnId {  get; set; } = 0;
}
